### PR TITLE
issue 1662 - remove the duplicated ones in the Makefile

### PR DESCRIPTION
**Pull request by xauthulei**
I am try to enhance the Makefile , see the issue #1662 ,  @linggao would you help review this, Thanks

### CODE REVIEW

File: image_bld.sh [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+		rm -rf ${WORK_DIR}/anax
+		rm -rf ${WORK_DIR}/hzn
+		cp ${EXECUTABLE} ${WORK_DIR}
+		cp ${CLI_EXECUTABLE} ${WORK_DIR}
+		cp -f ${LICENSE_FILE} ${WORK_DIR}
+	fi
+	cd ${WORK_DIR} && docker build ${DOCKER_MAYBE_CACHE} ${IMAGE_LABELS} -t ${IMAGE_NAME} -f Dockerfile.ubi.${arch} . && \
+	docker tag ${IMAGE_NAME} ${IMAGE_STG}
+else 
+	echo "Building the anax docker image is not supported on ${arch}"
```
Comment (bencourliss) [inline image_bld.sh:66]: Should replace anax with the actual component name.

File: promote_image.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
+#! /bin/bash
+
+#This scripts is used to push the images to repository
+
+IMAGE_NAME=$1
+IMAGE_PROD=$2
+IMAGE_LATEST=$3
+
+docker pull ${IMAGE}
```
Comment (bencourliss) [inline promote_image.sh:9]: You're not setting the `${IMAGE}` variable. I think you mean `${IMAGE_NAME}`

Comment (bencourliss) [reply promote_image.sh:9]: Likewise, I'd want more robustness around the arguments here including checking for valid choices and ensuring it errors out with an appropriate message if needed.

File: image_bld.sh [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+if [ "A${DOCKER_MAYBE_CACHE}" == "A" ];then
+	echo "Please define docker cache option."
+fi
+if [ "A${IMAGE_STG}" == "A" ];then
+	echo "Please define the image tag."
+fi
+
+# Special requirements for agbot, anax-k8s ,css-docker, ess-docker image build
+if [[ ${arch} != "amd64" && ${COMP_NAME} == "agbot" ]]; then 
+	echo "agbot image building just support on amd64 !!!"
```
Comment (bencourliss) [inline image_bld.sh:29]: Let's clean up the message to use correct English and punctuation on this one and the others below it. I'd suggest something like "Building the agbot image is only supported on amd64 architectures."

**Review by bencourliss — CHANGES_REQUESTED** [review]

File: image_bld.sh (hunk 3 above)
Comment (xauthulei) [reply image_bld.sh:29]: Great, I will replace it. Thanks

File: image_bld.sh [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+fi
+if [ "A${IMAGE_LABELS}" == "A" ];then
+	echo "Please define the image labels."
+fi
+if [ "A${DOCKER_MAYBE_CACHE}" == "A" ];then
+	echo "Please define docker cache option."
+fi
+if [ "A${IMAGE_STG}" == "A" ];then
+	echo "Please define the image tag."
+fi
```
[the remark this answers is not in the document]
Comment (xauthulei) [reply image_bld.sh:25]: Good ideas, I will replace it,Thanks @bencourliss

File: promote_image.sh (hunk 2 above)
Comment (xauthulei) [reply promote_image.sh:9]: @bencourliss you are right, it should be `${IMAGE}`, Thanks

File: image_bld.sh [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+
+# Special requirements for agbot, anax-k8s ,css-docker, ess-docker image build
+if [[ ${arch} != "amd64" && ${COMP_NAME} == "agbot" ]]; then
```
[the remark this answers is not in the document]
Comment (xauthulei) [reply image_bld.sh:28]: Initially, I want fetch it from the outside defined environment (it have been defined in the `Makefile`). Is that OK @bencourliss

**Comment by xauthulei** [comment]
@bencourliss , I have refreshed the code, could you help review this again. Many Thanks

**Pull request closed without merging**